### PR TITLE
Removed trailing whitespace when no service pack version exists

### PR DESCRIPTION
**Pull request by umberbeard**
When calling `RuntimeInformation.OSDescription` on Windows 10, there is trailing whitespace caused by an empty service pack version field in the underlying data structure from the Windows SDK `RtlGetVersion` call. This change aims to check whether the service pack string should actually be included.

### CODE REVIEW

File: src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs [hunk 1]
```diff
@@ -20,8 +19,19 @@ internal static unsafe string RtlGetVersion()
...
+                stringBuilder.Append(" ");
+                stringBuilder.AppendFormat("{0}.{1}.{2}", osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber);
+
+                if (osvi.szCSDVersion[0] != '\0')
+                {
+                    stringBuilder.Append(" ");
+                    stringBuilder.Append(new string(&(osvi.szCSDVersion[0])));
+                }
+
+                return stringBuilder.ToString();
```
Comment (stephentoub) [inline src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs]: How about just changing the existing:
```C#
return string.Format("{0} {1}.{2}.{3} {4}",
    version, osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber, new string(&(osvi.szCSDVersion[0])));
```
to instead be:
```C#
return string.Format("{0} {1}.{2}.{3}{4}",
    version,
    osvi.dwMajorVersion,
    osvi.dwMinorVersion,
    osvi.dwBuildNumber,
    osvi.szCSDVersion[0] != '\0' ? " " + new string(&(osvi.szCSDVersion[0])) : string.Empty);
```
?

Comment (umberbeard) [reply src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs]: I wrote exactly this code but wanted to check with someone on the .NET maintainers list. I spoke to Viktor Hofer who asked that I consider using a `StringBuilder` instead.

File: src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs [hunk 2]
```diff
@@ -2,9 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Runtime.InteropServices;
-using System.Security;
+using System.Text;
```
Comment (stephentoub) [inline src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs]: Is this using needed now?

Comment (umberbeard) [reply src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs]: Nope. Removed

File: src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs (hunk 1 above)
Comment (ViktorHofer) [reply src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs]: To be clear I recommended to either use string.Format, StringBuilder or Span if its a hot path :)

**Pull request merged**